crates/stackable-webhook/src/webhooks/validating_webhook.rs

```diff
@@ -0,0 +1,162 @@
+use std::{fmt::Debug, marker::PhantomData, sync::Arc};
+
+use async_trait::async_trait;
+use axum::{Json, Router, routing::post};
+use k8s_openapi::{ByteString, api::admissionregistration::v1::ValidatingWebhookConfiguration};
+use kube::{
+    Api, Client, Resource, ResourceExt,
+    api::{Patch, PatchParams},
+    core::admission::{AdmissionRequest, AdmissionResponse, AdmissionReview},
+};
+use serde::{Serialize, de::DeserializeOwned};
+use snafu::{ResultExt, Snafu};
+use tracing::instrument;
+use x509_cert::Certificate;
+
+use super::{Webhook, WebhookError};
+use crate::{WebhookServerOptions, webhooks::get_webhook_client_config};
+
+#[derive(Debug, Snafu)]
+pub enum ValidatingWebhookError {
+    #[snafu(display("failed to patch ValidatingWebhookConfiguration {vwc_name:?}"))]
+    PatchValidatingWebhookConfiguration {
+        source: kube::Error,
+        vwc_name: String,
+    },
+}
+
+/// Validating webhook, which let's you intercept object creations/modification and allow or deny
+/// the object.
+///
+/// As the webhook is typed with the Resource type `R`, it can only handle a single resource
+/// validation. Use multiple [`ValidatingWebhook`] if you need to validate multiple resource kinds.
+///
+/// ### Example usage
+///
+/// TODO
+pub struct ValidatingWebhook<H, S, R> {
+    validating_webhook_configuration: ValidatingWebhookConfiguration,
+    handler: H,
+    handler_state: Arc<S>,
+    _resource: PhantomData<R>,
+
+    disable_validating_webhook_configuration_maintenance: bool,
```

Comment:
suggestion: This is a mouthful with a bunch of superfluous information: The fact that we are talking about validation webhooks is already indicated by the struct name itself and the maintenance _is_ dealing with the validating webhook config.

```suggestion
    disable_maintenance: bool,
```